cli/tests/test_branch_command.rs

```diff
@@ -189,6 +189,31 @@ fn test_branch_move_conflicting() {
     "###);
 }
 
+#[test]
+fn test_branch_rename() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    let stderr = test_env.jj_cmd_failure(&repo_path, &["branch", "rename", "foo", "bar"]);
+    insta::assert_snapshot!(stderr, @r###"
+    Error: No such branch: foo
+    "###);
+
+    // we will conflict with this in a later test.
+    test_env.jj_cmd_ok(&repo_path, &["branch", "create", "conflictfoo"]);
```

Comment:
nit: move it just before the failing command? then you might not even need a comment

Reply:
Done.